lib/cobra_commander/component_tree/combined_component_tree.rb

```diff
@@ -0,0 +1,94 @@
+# frozen_string_literal: true
+
+require "oj"
+
+module CobraCommander
+  class ComponentTree
+    # Represents a tree of all component dependencies
+    class CombinedComponentTree
+      def initialize(name, path, ancestry = Set.new)
+        @name = name
+        @root_path = path
+        @ancestry = ancestry
+      end
+
+      def to_h
+        {
+          name: @name,
+          path: @root_path,
+          ancestry: @ancestry,
+          dependencies: dependencies.map(&method(:dep_representation)),
+        }
+      end
+
+      def node_tree
+        @node_tree ||= NodeComponentTree.new(UMBRELLA_APP_NAME, @root_path).to_h
+      end
+
+      def gem_tree
+        @gem_tree ||= GemComponentTree.new(UMBRELLA_APP_NAME, @root_path).to_h
+      end
+
+      def js_format(deps)
```

Comment:
Move this method down below more significant ones? I have to jump a long way between the definition and the usage.